### PR TITLE
Add Firewall/SSO type "JSON-RPC"

### PR DESCRIPTION
**Pull request by tribut**
# Description

This is essentially https://github.com/inverse-inc/packetfence/pull/1507 with the SSO module rewritten in Go and some documentation added. Sorry for letting the other PR rot, I was busy and shouldn't have opened the PR without the time to follow up. At least this way you did not have to figure out how to do JSON-RPC 2.0 over HTTPS in Go :nerd_face: 

The PR adds support for a new SSO type "JSONRPC". [We](https://www.stunet.tu-freiberg.de/) were looking to integrate Packetfence with our linux firewall, but unfortunately there is no "generic" firewall/SSO type yet and I wanted something as simple as possible. This is my proposed solution for this problem.

The works well with the equally trivial server from https://github.com/tribut/ipset-rpcd

Please let me know specifically if you think the Guide should be more specific. I decided against an explanation how to install `ipset-rpcd` because it is probably out-of-scope, but I'd let you make that call if you disagree.

# NEWS file entries
## New Features
* New Firewall/SSO type: JSON-RPC

### CODE REVIEW

**Comment by julsemaan** [comment]
First, thanks for your contribution.

Not really much to say about it, code is clean, doc is a bit short but anyway its aimed at very advanced users.

I'd be happy to merge this before the 7.1 merge window closes unless someone from the team sees something I missed

File: docs/PacketFence_JSON-RPC_Quick_Install_Guide.asciidoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+This guide has been created in order to help sales engineers, product managers, or network specialists demonstrate the PacketFence capabilities on-site with an existing or potential customer.  It can also provide guidelines to setup a proof of concept for a potential PacketFence deployment using a firewall with a *JSON-RPC* interface.
+
+Assumptions
+-----------
+* You have a configured PacketFence environment with working test equipment;
+* You have a firewall with a compatible JSON-RPC interface (see below).
+
+JSON-RPC interface
+------------------
+The JSONRPC module shipped with packetfence is meant as a generic firewall SSO module to be used with Linux or BSD firewalls that do not by default ship with a vendor-specific interface to do SSO with.
```
Comment (extrafu) [inline docs/PacketFence_JSON-RPC_Quick_Install_Guide.asciidoc]: packetfence -> PacketFence

Comment (tribut) [reply docs/PacketFence_JSON-RPC_Quick_Install_Guide.asciidoc]: Thanks. I've also updated the spelling further down the same document.

**Comment by dwlfrth** [comment]
@whitx comment

File: docs/PacketFence_JSON-RPC_Quick_Install_Guide.asciidoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  - `role`    _(string)_: PacketFence role assigned to the device
+  - `timeout` _(int)_:    Duration until the registration expires in seconds
+* *Response*: Success must be indicated by `"result": ["OK"]`. Every string other than `OK` is taken as an error message.
+
+A simple JSON-RPC server written in Python that is compatible with this specification and creates ipsets based on the SSO information provided by PacketFence can be found at https://github.com/tribut/ipset-rpcd.
+
+SSO Configuration in PacketFence
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Go to *'Configuration' -> 'Firewall SSO' -> 'Add Firewall' -> 'JSONRPC'*.
```
Comment (whitx) [inline docs/PacketFence_JSON-RPC_Quick_Install_Guide.asciidoc]: The path should be "Configuration -> Integration -> Firewall SSO -> ..." (from 7.0 and onward)

Comment (tribut) [reply docs/PacketFence_JSON-RPC_Quick_Install_Guide.asciidoc]: Sorry, must have missed that. Fixed.

**Pull request merged**